nightmare/src/main/java/net/runelite/client/plugins/nightmare/NightmarePlugin.java

```diff
@@ -0,0 +1,298 @@
+		{
+			cursed = false;
+		}
+
+	}
+
+	@Subscribe
+	private void onConfigChanged(ConfigChanged event)
+	{
+		if (!event.getGroup().equals("betterNightmare"))
+		{
+			return;
+		}
+		this.updateConfig();
+	}
+
+	@Subscribe
+	public void onAnimationChanged(AnimationChanged event)
+	{
+		if (!inFight || nm == null)
+		{
+			return;
+		}
+
+		Actor actor = event.getActor();
+		if (!(actor instanceof NPC))
+		{
+			return;
+		}
+
+		NPC npc = (NPC) actor;
+		int id = npc.getId();
+		int animationId = npc.getAnimation();
+
+		if (animationId == NIGHTMARE_MAGIC_ATTACK)
+		{
+			ticksUntilNextAttack = 7;
+			attacksSinceCurse++;
+			pendingNightmareAttack = cursed ? NightmareAttack.CURSE_MAGIC : NightmareAttack.MAGIC;
+		}
+		else if (animationId == NIGHTMARE_MELEE_ATTACK)
+		{
+			ticksUntilNextAttack = 7;
+			attacksSinceCurse++;
+			pendingNightmareAttack = cursed ? NightmareAttack.CURSE_MELEE : NightmareAttack.MELEE;
+		}
+		else if (animationId == NIGHTMARE_RANGE_ATTACK)
+		{
+			ticksUntilNextAttack = 7;
+			attacksSinceCurse++;
+			pendingNightmareAttack = cursed ? NightmareAttack.CURSE_RANGE : NightmareAttack.RANGE;
+		}
+		else if (animationId == NIGHTMARE_CURSE)
+		{
+			cursed = true;
+			attacksSinceCurse = 0;
+		}
+
+		if (cursed && attacksSinceCurse == 5)
+		{
+			//curse is removed when she phases, or does 5 attacks
+			cursed = false;
+			attacksSinceCurse = -1;
+		}
+	}
+
+	@Subscribe
+	public void onNpcDefinitionChanged(NpcDefinitionChanged event)
+	{
+		final NPC npc = event.getNpc();
+
+		if (npc == null)
+		{
+			return;
+		}
+
+		//this will trigger once when the fight begins
+		if (npc.getId() == 9432)
+		{
+			//reset everything
+			reset();
+			nm = npc;
+			inFight = true;
+		}
+
+		//if ID changes to 9431 (3rd phase) and is cursed, remove the curse
+		if (cursed && npc.getId() == 9431)
+		{
+			cursed = false;
+			attacksSinceCurse = -1;
+		}
+
+		//if npc is in the totems map, update its phase
+		if (totems.containsKey(npc.getIndex()))
+		{
+			totems.get(npc.getIndex()).updateCurrentPhase(npc.getId());
+		}
+		else if (INACTIVE_TOTEMS.contains(npc.getId()))
+		{
+			//else if the totem is not in the totem array and it is an inactive totem, add it to the totem map.
+			totems.putIfAbsent(npc.getIndex(), new MemorizedTotem(npc));
+		}
+	}
+
+	@Subscribe
+	private void onGameTick(final GameTick event)
+	{
+		if (!inFight || nm == null)
+		{
+			return;
+		}
+
+		//if nightmare's id is 9433, the fight has ended and everything should be reset
+		if (nm.getId() == 9433)
+		{
+			reset();
+		}
+
+		ticksUntilNextAttack--;
+
+		if (pendingNightmareAttack != null && ticksUntilNextAttack == 0)
+		{
+			pendingNightmareAttack = null;
+		}
+	}
+
+	private boolean isNightmareNpc(int id)
+	{
+		return id >= 9425 && id <= 9433;
+	}
+
+	private void updateConfig()
+	{
+		this.prayerHelper = config.prayerHelper();
+		this.tickCounter = config.ticksCounter();
+		this.highlightTotems = config.highlightTotems();
+	}
```

Comment:
we have done away with this method, just use config.bla when using config instead of this.bla

Reply:
changes made in latest commit